### PR TITLE
Fix 3257: add feature to send mail when host creates a new challenge

### LINKED ISSUE #3257: [Backend] Send email with challenge pk and queue name for a remote evaluation challenge
### Description

For remote evaluation challenge, challenge hosts need `challenge_pk` and challenge `queue_name` to start a remote evaluation worker. When a challenge host creates a new challenge we should send an email with `challenge_pk` and `queue_name`

### PR DESCRIPTION
**Pull request by burnerlee**
resolves #3257 by sending a mail for remote evaluation when a new challenge is created by a host user

### CODE REVIEW

**Comment by burnerlee** [comment]
@Ram81 ping for review

File: apps/challenges/models.py [hunk 1]
```diff
@@ -202,6 +203,8 @@ def create_eks_cluster_for_challenge(sender, instance, created, **kwargs):
         ):
             serialized_obj = serializers.serialize("json", [instance])
             aws.setup_eks_cluster.delay(serialized_obj)
+    if created:
+        construct_and_send_remote_evaluation_mail(instance)
```
Comment (Ram81) [inline apps/challenges/models.py:207]: @burnerlee we should call this method from `create_challenge_using_zip` API in `views/challenges.py`. When a challenge is successfully uploaded using the API and challenge is a remote evaluation challenge then only we will send this email. You can check if a challenge is a remote_evaluation challenge using `remote_evaluation` field from challenge model

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Comment by burnerlee** [comment]
@Ram81 changes made

**Comment by burnerlee** [comment]
@Ram81 ping

**Comment by Ram81** [comment]
@burnerlee here is the template id `d-0e5c6256407e4f9baa5a3f51c95af435`

File: apps/challenges/challenge_notification_util.py [hunk 2]
```diff
@@ -56,3 +56,21 @@ def construct_and_send_eks_cluster_creation_mail(challenge):
...
         template_data=template_data,
     )
+
+
+def construct_and_send_remote_evaluation_mail(challenge):
+    if settings.DEBUG:
+        return
+
+    template_data = {"CHALLENGE_PK": challenge.id,
+                     "QUEUE_NAME": challenge.queue}
```
Comment (Ram81) [inline apps/challenges/challenge_notification_util.py:66]: @burnerlee change `QUEUE_NAME` to `CHALLENGE_QUEUE_NAME`.

File: apps/challenges/challenge_notification_util.py [hunk 3]
```diff
@@ -56,3 +56,21 @@ def construct_and_send_eks_cluster_creation_mail(challenge):
         template_id=template_id,
         template_data=template_data,
     )
+
+
+def construct_and_send_remote_evaluation_mail(challenge):
+    if settings.DEBUG:
+        return
+
+    template_data = {"CHALLENGE_PK": challenge.id,
```
Comment (Ram81) [inline apps/challenges/challenge_notification_util.py:65]: @burnerlee we also need to send `CHALLENGE_NAME`, `CHALLENGE_IMAGE_URL` and `CHALLENGE_URL` to the email template

File: apps/challenges/challenge_notification_util.py [hunk 4]
```diff
@@ -56,3 +56,21 @@ def construct_and_send_eks_cluster_creation_mail(challenge):
         template_id=template_id,
         template_data=template_data,
     )
+
+
+def construct_and_send_remote_evaluation_mail(challenge):
```
Comment (Ram81) [inline apps/challenges/challenge_notification_util.py:61]: Please rename to `construct_and_send_challenge_details_email`

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Comment by burnerlee** [comment]
@Ram81, where do i get the url of the challenge from? There seems to be no field for this is challenge model

**Comment by Ram81** [comment]
@burnerlee see the method `construct_and_send_worker_start_mail` in `challenges/challenge_notification_util.py`. Challenge URL is not stored in models it is constructed on the fly.

**Comment by burnerlee** [comment]
@Ram81 thank you for the help, made the changes

File: apps/challenges/aws_utils.py [hunk 5]
```diff
@@ -1094,11 +1094,15 @@ def create_eks_nodegroup(challenge, cluster_name):
         response = client.create_nodegroup(
             clusterName=cluster_name,
             nodegroupName=nodegroup_name,
-            scalingConfig={"minSize": 1, "maxSize": 10, "desiredSize": 1},
-            diskSize=100,
+            scalingConfig={
+                "minSize": challenge_obj.min_worker_instance,
+                "maxSize": challenge_obj.max_worker_instance,
```
Comment (Ram81) [inline apps/challenges/aws_utils.py:1099]: @burnerlee why are these changes part of your PR?

File: apps/challenges/models.py [hunk 6]
```diff
@@ -152,6 +152,21 @@ def __init__(self, *args, **kwargs):
     subnet_2_cidr = models.CharField(
         null=True, blank=True, max_length=200, default=""
     )
+    # Evaluation instance config for code upload challenge
+    worker_instance_type = models.CharField(
```
Comment (Ram81) [inline apps/challenges/models.py:156]: @burnerlee these changes should not be part of your PR. Please revert them

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Comment by burnerlee** [comment]
Maybe I pulled some wrong branch into my branch and all this is messed up, closing this PR and raisinhhg another one with the required changes from a new branch

**Pull request closed without merging**